### ISSUE TITLE
Add authentication to server application

### ISSUE THREAD

**Issue by autumnfound** [EclipseFdn/react-eclipsefdn-members#9]
To better protect user data and ensure that users can only access information about filled forms if they are logged in we should make sure that authentication is used on both ends of the stack. In the Web UI, we need to trigger a login on the server and be able to handle the logged-in state and detect it. The UI should have no interaction with user credentials and session tokens, only requesting for user data as a litmus to authentication state.

The server needs to be able to create secure login sessions for a user that is transparent to the front end, and secure from XSS and data mining vulnerabilities. While a key could be stored in cookies to fingerprint a session, there should be no sensitive data in the key, and we should make sure that there are other security measures around sessions.

The current effort uses Quarkus built-in security modules as a way of handling authentication sessions in Keycloak. It seems to tick all of the boxes we need for usage in the server and allows for easy login through redirects.

**Comment by chrisguindon** [comment]
I agree that we need authentication on both the front and back-end. However, I would like to focus on the backend with this issue since that work is needed before we can implement this in the front-end.

I will create a separate issue to track the front-end.

**Comment by chrisguindon** [comment]
Additional information about these implementation details are available here: 
https://docs.google.com/document/d/1DAdolxLQrOAUqFSsBge5StGLAtTcMMIt-vfzumlnatY/edit#heading=h.1wj3ctmandot

We experimented with two solutions:

- Solution 1: Authorization code grant managed by a dynamic server back-end
- Solution 2: JWT Bearer token authorization grant type for OAuth 2.0

We concluded that the best way to securely implement OAuth in a JavaScript app is to keep the token management outside of JavaScript entirely :)

**Comment by chrisguindon** [comment]
Instead of creating an issue now about the front-end, I would like to make it a requirement for this issue.

I think once we made progress here, we will be in a better position to define the front-end requirements for YI.

**Comment by chrisguindon** [comment]
This is done.
